lib/logstash/devutils/rspec/shared_examples.rb

```diff
@@ -0,0 +1,23 @@
+require 'rspec/wait'
+
+RSpec.shared_examples "an interruptible input plugin" do
+  describe "#stop" do
+    let(:queue) { SizedQueue.new(20) }
+    subject do
+      s = described_class.new(config)
```

Comment:
I don't think you can do that. What i think might work for people is
redefining subject.

/purbon

On Wed, 16 Sep 2015 12:36 João Duarte <notifications@github.com> wrote:

> In lib/logstash/devutils/rspec/shared_examples.rb
> <https://github.com/elastic/logstash-devutils/pull/32#discussion_r39614914>
> :
>
> > @@ -0,0 +1,23 @@
> > +require 'rspec/wait'
> > +
> > +RSpec.shared_examples "an interruptible input plugin" do
> > +  describe "#stop" do
> > +    let(:queue) { SizedQueue.new(20) }
> > +    subject do
> > +      s = described_class.new(config)
>
> I see the point, my purpose with this was to reduce the overhead of using
> this shared_example everywhere. I wonder if there's a way to make
> described_class.new(config) the default, but allow overriding this
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/elastic/logstash-devutils/pull/32/files#r39614914>.
>